lib/spack/llnl/util/symlink.py

```diff
@@ -9,59 +9,136 @@
 from os.path import exists, join
 from sys import platform as _platform
 
+import llnl.util.tty as console
 from llnl.util import lang
 
 is_windows = _platform == "win32"
 
 if is_windows:
+    import subprocess
+
     from win32file import CreateHardLink
 
 
 def symlink(real_path, link_path):
```

Comment:
Could we also start calling this "link" since this is no-longer just for creating symlinks?

(I'll note this is not an issue introduced by this PR, since it was creating non-symlinks before, but it would be good to change here)

Reply:
I would recommend leaving this for another PR. There are many uses of symlink in the code that are alongside calls to os.symlink in the same files, so it would be complicated change to make wholesale. It seems like we should slowly change the references to os.symlink over to llnl.util.symlink and then we can make a change of name with less impact.